src/isec_surf_surf.py

```diff
@@ -1,4 +1,249 @@
+import sys
+build_path="/home/jiri/Soft/Geomop/Intersections/external/bih/build"
+sys.path+=[build_path]
+print(sys.path)
+
+import bih
+import numpy as np
+import numpy.linalg as la
+
+
+
+
+
 class IsecSurfSurf:
+
+    def __init__(self, surf1, surf2, nt=2,nit=10):
+        self.surf1 = surf1
+        self.surf2 = surf2
+        self.tree1 = self.bounding_boxes(self.surf1)
+        self.tree2 = self.bounding_boxes(self.surf2)
+        self.nt = nt
+        self.nit = nit
+
+    @staticmethod
+    def bounding_boxes(surf):
+        tree = bih.BIH()
+        n_patch = (surf.u_basis.n_intervals)*(surf.v_basis.n_intervals)
+
+
+        patch_poles = np.zeros([9, 3, n_patch])
+        i_patch = 0
+        for k in range(surf.u_basis.n_intervals):
+            for l in range(surf.v_basis.n_intervals):
+                n_points = 0
+                for i in range(0,3):
+                    for j in range(0,3):
+                        patch_poles[n_points,:,i_patch] = surf.poles[k+i, l+j, :]
+                        n_points += 1
+                i_patch += 1
+        boxes = [bih.AABB(patch_poles[:,:,p].tolist()) for p in range(n_patch)]
+        #print(patch_poles[:, :, 0])
+        tree.add_boxes( boxes )
+        tree.construct()
+        #print(boxes)
+        #for b in boxes:
+        #    print(b.min()[0:2],b.max()[0:2])
+        return boxes, tree
+
+
+
+    @staticmethod
+    def get_intersection(surf1,surf2,tree1,tree2,box1,box2,nt,nit):
```

Comment:
rename - compute
remove static, remove parameters these are in the object i.e. 'self'